src/module-config/module-config.ts

```diff
@@ -50,15 +54,15 @@ export async function getConfig(moduleName: string): Promise<any> {
   // of the provided properties exist in the schema.
   const checkForUnknownConfigProperties = (schema, config, keyPath = "") => {
     for (let [key, value] of Object.entries(config)) {
-      keyPath += key;
+      const thisKeyPath = keyPath + key;
```

Comment:
This appears to be a separate bug from the infinite loop, no? Just that we couldn't really handle multiple configs since we were looking up the wrong keyPath after the first iteration of this for loop?

Reply:
Yeah, separate bug. It's only actually used for the error message, though. The only problem it was causing is that the error message would read "Unknown config key 'foobarbaz'..." instead of "Unknown config key 'foo.bar.baz'...".